packages/@angular/cli/tasks/serve.ts

```diff
@@ -31,7 +31,15 @@ export default Task.extend({
     if (projectConfig.project && projectConfig.project.ejected) {
       throw new SilentError('An ejected project cannot use the build command anymore.');
     }
-    rimraf.sync(path.resolve(this.project.root, outputPath));
+
+    const contentBase = path.resolve(this.project.root, outputPath);
+    if (serveTaskOptions.useDist) {
+      if(!fs.existsSync(contentBase)) {
+        throw new SilentError('Dist directory not found');
```

Comment:
Yes the point of this is exactly to serve a previous build. It allows the user the option to build using whatever options they want with the files written to disk and serve that. More importantly, it allows running e2e's against a build for use on a build machine / continuous deployment. Today to deploy using ng-cli I'm building the site 3 times.

- `ng test` builds and then tests
- `ng e2e` runs ng serve and e2e's
- `ng build` to create the actual bits for deployment

I'd rather use `ng build` first to build EXACTLY what will be deployed then run the unit tests and e2e tests against what will be deployed. This is a pretty standard thing to do.